pilot/pkg/networking/util/util.go

```diff
@@ -614,6 +614,35 @@ func StringToPrefixMatch(in []string) []*matcher.StringMatcher {
 	return res
 }
 
+// StringToExactMatchWithFallback creates prefix matchs as fallbacks in addition to the
+// converted exact matches, if string inputs end with the wildcard.
+func StringToExactMatchWithFallback(in []string) []*matcher.StringMatcher {
+	if len(in) == 0 {
+		return nil
+	}
+
+	res := make([]*matcher.StringMatcher, 0, len(in))
+	for _, s := range in {
+		// Replace e.g. "foo" with a exact match
+		res = append(res, &matcher.StringMatcher{
+			MatchPattern: &matcher.StringMatcher_Exact{Exact: s},
+		})
+
+		// Replace e.g. "foo.*" with prefix match
```

Comment:
My memory here is a bit fuzzy and I didn't dig up the RFC, but my memory of SAN matching is:
* Prefix match is not allowed
* Suffix match only goes one level deep. So `*.foo.com` should not match `a.b.foo.com`. There was some envoy CVE about this

Reply:
Thanks for the summary! They're mostly correct IMO (though not clear/consistent amongst different impl.).

However, usually this only applies to `subjectAltName`s of type DNS name (or similar). For an ID of type `application service`, e.g., SPIFFE URI, there seems to be no such standard.

https://tools.ietf.org/html/rfc6125#section-6 might be a good reference.

Reply:
Yeah that makes sense, but this API can be used for either so its a bit tricky. We need to be careful about being so permissive that we allow people to do things that are not safe. 

@myidpt @costinm may have some opinions here